src/rest/SearchEndpoint.ts

```diff
@@ -382,6 +383,42 @@ export class SearchEndpoint implements ISearchEndpoint {
       return results;
     });
   }
+
+  /**
+   * Gets the plan of execution of a search, without performing it.
+   *
+   * @param query The query to execute. Typically, the query object is built using a
+   * [`QueryBuilder`]{@link QueryBuilder}.
+   * @param callOptions An additional set of options to use for this call.
+   * @param callParams The options injected by the applied decorators.
+   * @returns {Promise<IPlanResults>} A Promise of plan results.
+   */
+  @path('/plan')
+  @method('POST')
+  @requestDataType('application/json')
+  @responseType('json')
+  @includeActionsHistory()
+  @includeReferrer()
+  @includeVisitorId()
+  @includeIsGuestUser()
```

Comment:
Since this call is pretty much supports all search() parameters, it would be better that they both use the same private "build" method (which use the same parameters except `path`|`responseType`|`requestDataType`.

Another approach could be to have one decorator that combines those in common with search(). 

So it would call all the other needed decorators in order: actionsHistory, referrer, visitorID, guestUser